### ISSUE TITLE
lodash vs. lodash/curry discrepancy 

### ISSUE THREAD

**Issue by billyjanitsch** [lodash/lodash#1955]
The `curry` bundled with the full (npm) Lodash behaves differently than the one in `lodash/curry`:

``` js
const curry1 = require('lodash').curry
const curry2 = require('lodash/curry')

const f = (a, b, c) => a + b + c

const f1 = curry1(f)
const f2 = curry2(f)

f1(1, undefined, 3) // => NaN
f2(1, undefined, 3) // => [Function: wrapper]
```

**Comment by jdalton** [comment]
This has been fixed in the `lodash-cli` side. It's due to the `placeholder` not getting set so it's `undefined` which makes the modular form think that `undefined` is a placeholder.

A temporary workaround is to 
```js
const curry2 = require('lodash/curry')
curry2.placeholder = {};
```

**Comment by jdalton** [comment]
It will be fixed on our end on the v4.3.0 update.
